### ISSUE TITLE
Pokemon listed in "evolve_before_transfer" does not evolve

### ISSUE THREAD

**Issue by casper-gh** [jabbink/PokemonGoBot#1204]
**Description:**
Pokemon listed in "evolve_before_transfer" does not evolve, instead it keeps all those pokemon causing `POKEMON_INVENTORY_FULL`

**Steps to reproduce:**

1. Add pokemon to `evolve_before_transfer`
2. Run the bot
3. All pokemon listed in `evolve_before_transfer` are kept causing `POKEMON_INVENTORY_FULL`

**Expected behavior:**
Pokemon should be evolved and then transferred

**Actual behavior:**
Pokemon does not evolve

**Version:**
0.6.0

**Operating System:**
Ubuntu

**Java version:**
Latest

Leave this sentence in your issue as proof that you have read and used this template.

My config: http://pastebin.com/GC8Ggh7u

**Comment by FrancYescO** [comment]
It willl wait for
evolve_stack_limit=10 before start evolving

**Comment by casper-gh** [comment]
@FrancYescO I don't really understand what it means, but I have like 30+ Pidgeys and bot still doesn't evolve

**Comment by FrancYescO** [comment]
Sometime it will log "Pokémon ready for stack evolve xx" probably you need more candys?

**Comment by casper-gh** [comment]
I got Pokémon ready for stack evolve 0, but I have 5K + of Pidgey candies

**Comment by FrancYescO** [comment]
Pidgy is also here..

obligatory_transfer=DODUO,RATTATA,CATERPIE,PIDGEY

**Comment by casper-gh** [comment]
but doesn't `evolve_before_transfer` will evolve the pokemon first? So what you're saying is I cannot define the same pokemon in both `obligatory_transfer` and `evolve_before_transfer` ?

**Comment by FrancYescO** [comment]
How can a Pokémon be evolved if you "obligatory" throw away?

**Comment by casper-gh** [comment]
The thing is, the bot neither throws those pokemon away nor evolve it, all those pokemon are retained making my pokemon inventory full

**Comment by FrancYescO** [comment]
Post your Pokémon list

**Comment by casper-gh** [comment]
Sorry I manually evolved them all, will try to get the list when pokemon are full again

**Comment by casper-gh** [comment]
Ok I got the pokemon list here http://pastebin.com/87Y8RrcJ

You can see there's a lot of Pidgeys...

**Comment by FrancYescO** [comment]
and still Pokémon ready for stack evolve 0 ???? if you don't see this incrementing and reaching 10 (as you config require) you will never get an evolve.

**Comment by casper-gh** [comment]
that's why it's a bug?

**Comment by FrancYescO** [comment]
yeah but to me is working good, so wee need to find what's wrong with you.

**Comment by casper-gh** [comment]
can you share your config?

**Comment by FrancYescO** [comment]
>   "obligatoryTransfer" : [ ],
  "evolveBeforeTransfer" : [ "CATERPIE", "RATTATA", "WEEDLE", "PIDGEY", "ZUBAT" ],
  "evolveStackLimit" : 70,
  "useLuckyEgg" : 1,

butw probably somehow related to
https://github.com/jabbink/PokemonGoBot/issues/1088

**Comment by casper-gh** [comment]
So don't put anything in obligatory transfer?

**Comment by FrancYescO** [comment]
no, i don't want to obligatory transfer.

**Comment by FrancYescO** [comment]
also make sure to edit the json file if present and not proprieties one.

**Comment by casper-gh** [comment]
can you share the entire json template?

**Comment by FrancYescO** [comment]
i'm using develop with some pr merged so some config will be in conflict.

what can affect this issue is already up

**Comment by MacDogg** [comment]
Having the same issue.
Config info related to this:
obligatory_transfer=CATERPIE,RATTATA,WEEDLE,PIDGEY **_(this seems to be ignored because none of these are transferred, they are all still in my inventory)_**
evolve_before_transfer=CATERPIE,RATTATA,WEEDLE,PIDGEY
evolve_stack_limit=30
use_lucky_egg=1
I currently have 14 Weedle, 15 Rattata, 23 Pidgey and 9 Caterpie.  When I start the bot with the above settings nothing happens, they aren't transferred, they aren't evolved.
I have 582 Weedle candy so I can evolve 48.  I have 942 Rattata candy so I can evolve 37.  I have 1256 Pidgey candy so I can evolve 104.  I have 297 Catepie candy so I can evolve 24.

**Comment by casper-gh** [comment]
in my case, the config on `obligatory_transfer` works fine, just `evolve_before_transfer` isn't working

**Comment by FrancYescO** [comment]
the crucial thing is 

> "evolveStackLimit" : 70,

no evolve are done **until we have enought pokemon to evolve** defined by this value

**obligatory_transfer pokemon are kept if also in the evolve_before_transfer** (still a nosense to have in both list)

**Comment by casper-gh** [comment]
My new config looks like this:

obligatory_transfer=DODUO
evolve_before_transfer=CATERPIE,RATTATA,WEEDLE,PIDGEY,DROWZEE

But after 8 hrs botting, still getting full pokemon inventory because all the above pokemon are kept.

**Comment by FrancYescO** [comment]
if in your log you see 

> Stack of pokemon ready to evolve: **0**

this is the issue but this can also mean you have not enought candy..

**Comment by MacDogg** [comment]
Damn, now I am stuck with almost a full inventory of Pokemon, none will transfer now after changing all the settings back to transfer everything under 90% IV.
autotransfer=false
keep_pokemon_amount=0
max_pokemon_amount=0
transfer_iv_threshold=90
transfer_cp_threshold=-1
transfer_cp_min_threshold=-1
ignored_pokemon=
obligatory_transfer=DODUO,CATERPIE,RATTATA,WEEDLE,PIDGEY
evolve_before_transfer=

**Comment by th3n3rd** [comment]
I had the same issue with Drowzee, like 1k+ candies and fill up the inventory. I've also tried to put it in the obligatory transfer and also in the list of "evolve before transfer" but I got into another issue where I got an arithmetic exception (division by zero). I believe there is some problem with the candies evaluation. I am using 0.6 by the way.

**Comment by FrancYescO** [comment]
@all confirm that:
 **"autotransfer" : true,**

**Comment by casper-gh** [comment]
Yes, that's what I have

**Comment by th3n3rd** [comment]
One thing that helps, just tried, is to use the `max_pokemon_amount` parameter. If you set the latte to 1 the bot will discard all the duplicate leaving only one pokemon per type but for the ones listed within `evolve_before_transfer` list! (unless you need candies of course).

**Comment by Sieberkev** [comment]
Can you get the most recent develop version (and not 0.6.0 release) and test it again? Several pieces of code for the evolving got improved.

**Comment by casper-gh** [comment]
I'm waiting for the release build, anyway I think the issue mostly happens if the bot is scheduled to restart periodically.

**Comment by kiaph** [comment]
Question:

details.

Player want only 3 of each pokemon max.
Players wants to evolve at stack 30
Player obtains 4 pidgey, does bot throw away pidgey, or keep it knowing it is basically going to be transfered?

Realistically the player keeps it as it is a "transfer pokemon", so hoping the bot is able to see this as well.

**Comment by FrancYescO** [comment]
Will keep it.
